src/test/java/seedu/address/logic/parser/SearchPublicAddressCommandParserTest.java

```diff
@@ -3,28 +3,31 @@
 import static seedu.address.logic.parser.CliSyntax.PREFIX_PUBLIC_ADDRESS;
 import static seedu.address.logic.parser.CommandParserTestUtil.assertParseFailure;
 import static seedu.address.logic.parser.CommandParserTestUtil.assertParseSuccess;
+
 import org.junit.jupiter.api.Test;
+
 import seedu.address.logic.commands.SearchPublicAddressCommand;
 public class SearchPublicAddressCommandParserTest {
-    private SearchPublicAddressCommandParser parser = new SearchPublicAddressCommandParser();
-    private final String nonEmptyPublicAddress = "Some public address.";
+    private final SearchPublicAddressCommandParser parser = new SearchPublicAddressCommandParser();
+
     @Test
-    public void parse_indexSpecified_success() {
+    public void parse_publicAddressSpecified_success() {
         // have public address
-        String userInput =  PREFIX_PUBLIC_ADDRESS + nonEmptyPublicAddress;
-        SearchPublicAddressCommand expectedCommand = new SearchPublicAddressCommand (nonEmptyPublicAddress);
-        assertParseSuccess(parser, userInput, expectedCommand);
-        // no public address
-        userInput = PREFIX_PUBLIC_ADDRESS+ "";
-        expectedCommand = new SearchPublicAddressCommand( "");
+        String nonEmptyPublicAddress = "Some public address.";
+        String userInput = SearchPublicAddressCommand.COMMAND_WORD + " "
+                + PREFIX_PUBLIC_ADDRESS + nonEmptyPublicAddress;
+        SearchPublicAddressCommand expectedCommand = new SearchPublicAddressCommand(nonEmptyPublicAddress);
         assertParseSuccess(parser, userInput, expectedCommand);
+
+
     }
     @Test
     public void parse_missingCompulsoryField_failure() {
-        String expectedMessage = String.format(MESSAGE_INVALID_COMMAND_FORMAT, SearchPublicAddressCommand.MESSAGE_USAGE);
+        String expectedMessage = String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                SearchPublicAddressCommand.MESSAGE_USAGE);
         // no parameters
         assertParseFailure(parser, SearchPublicAddressCommand.COMMAND_WORD, expectedMessage);
-        // no public address
+        // no parameters + space
```

Comment:
Instead of failing when there's a space, might be better to strip surrounding whitespaces like other commands